### PR TITLE
Feature upload dataset

### PR DESCRIPTION
**Pull request by shweta2101**
Completed the following: 
- Added stepper UI component. 
- Complete UI and API integration for upload dataset implemented.
- Added validations step wise.

Screenshots : 
Screenshot 2024-03-25 at 7 48 23 PM
Screenshot 2024-03-26 at 7 06 52 PM
Screenshot 2024-03-28 at 6 29 46 PM

Screenshot 2024-03-28 at 6 29 52 PM

### CODE REVIEW

File: tdei-ui/src/services/apiServices.js [hunk 1]
```diff
@@ -187,3 +187,24 @@ export async function postCreateStation(data) {
...
+  formData.append('tdei_project_group_id', data[0].tdei_project_group_id);
+  formData.append('tdei_service_id', data[0].tdei_service_id);
+  formData.append('dataset', data[1]);
+  formData.append('metadata', data[2]);
+  if (data[3] != null) {
+    formData.append('changeset', data[3]);
+  }
+  console.log('formData-tdei_project_group_id:', formData.get('tdei_project_group_id'));
+  console.log('formData-tdei_service_id:', formData.get('tdei_service_id'));
+  console.log('formData-dataset:', formData.get('dataset'));
```
Comment (MashB) [inline tdei-ui/src/services/apiServices.js:201]: remove consoles logs from all locations

File: tdei-ui/src/services/apiServices.js [hunk 2]
```diff
@@ -187,3 +187,24 @@ export async function postCreateStation(data) {
...
+  console.log('formData-metadata:', formData.get('metadata'));
+  const response = await axios.post(`https://tdei-api-dev.azurewebsites.net/api/v1/osw/upload/${data[0].tdei_project_group_id}/${data[0].tdei_service_id}`, formData, {
```
Comment (MashB) [inline tdei-ui/src/services/apiServices.js:203]: dont hard code urls , make it env variable

File: tdei-ui/package.json [hunk 3]
```diff
@@ -20,11 +25,14 @@
...
     "react-copy-to-clipboard": "^5.1.0",
     "react-dom": "^18.2.0",
+    "react-dropzone": "^14.2.3",
     "react-intersection-observer": "^9.4.1",
     "react-query": "^3.39.2",
     "react-redux": "^8.0.5",
     "react-router-dom": "^6.6.1",
     "react-scripts": "5.0.1",
+    "react-select": "^5.8.0",
+    "react-toastify": "^10.0.5",
```
Comment (MashB) [inline tdei-ui/package.json:35]: since we are using bootstrap and now added MUI , dont they have select and toast components ?

Comment (shweta2101) [reply tdei-ui/package.json:35]: They do, but I have seen a requirement where toast has to dismiss after 30 seconds. This toast by default provides it. Also, easily customizable.

File: tdei-ui/src/services/apiServices.js (hunk 1 above)
Comment (shweta2101) [reply tdei-ui/src/services/apiServices.js]: Resolved.

File: tdei-ui/src/services/apiServices.js (hunk 2 above)
Comment (shweta2101) [reply tdei-ui/src/services/apiServices.js]: Resolved.

File: tdei-ui/package.json (hunk 3 above)
Comment (shweta2101) [reply tdei-ui/package.json]: Resolved and replaced with bootstrap toast. 
Screenshot 2024-03-28 at 6 10 31 PM

Screenshot 2024-03-28 at 6 24 58 PM

**Review by susrisha — APPROVED** [review]

**Pull request merged**